src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3595,6 +3597,11 @@ private void CheckDisallowedNullAssignment(TypeWithState state, FlowAnalysisAnno
 
                         // track state by assigning from a fictional value from the parameter to the argument.
                         var parameterValue = new BoundParameter(argument.Syntax, parameter);
+
+                        // If the LHS has annotations, we perform an additional check for nullable value types
+                        CheckDisallowedNullAssignment(parameterWithState, leftAnnotations, argument.Syntax.Location);
+
+                        AdjustSetValue(argument, declaredType, lValueType, ref parameterWithState);
```

Comment:
Is there a test exercising this change? And is there a similar change needed for `case RefKind.Ref`?

Reply:
Yes, this is exercised by the deconstruction test. For `(Property1, Property2) = expr;` this simulates something like `expr.Deconstruct(out Property1, out Property2)`, even though you could not write that directly (not legal C#).
However, there is no scenario that does something like `M(ref Property)` as far as I can tell, so `ref` case is untouched.

---
In reply to: [323002797](https://github.com/dotnet/roslyn/pull/38610#discussion_r323002797) [](ancestors = 323002797)

Reply:
ref-returning properties can be passed by ref, but I don't know what expectations we have around the flow state of such return values.

Reply:
Does this comment still need attention? Feel free to mark resolved if you feel the important scenarios for now are addressed.

---
In reply to: [323443357](https://github.com/dotnet/roslyn/pull/38610#discussion_r323443357) [](ancestors = 323443357)